src/lib/planningApplication/converter.ts

```diff
@@ -0,0 +1,326 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Digital Planning Register is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Digital Planning Register. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import { DprApplication, DprPlanningApplication } from "@/types";
+import { PostSubmissionMetadata } from "@/types/odp-types/schemas/postSubmissionApplication/Metadata";
+import { getApplicationDprDecisionSummary } from "./decision";
+import { getApplicationDprStatusSummary } from "./status";
+import { ProcessStage } from "@/types/odp-types/schemas/postSubmissionApplication/enums/ProcessStage";
+import { ApplicationStatus } from "@/types/odp-types/schemas/postSubmissionApplication/enums/ApplicationStatus";
+import { AssessmentDecision } from "@/types/odp-types/schemas/postSubmissionApplication/enums/AssessmentDecision";
+import { getPrimaryApplicationTypeKey } from "./type";
+import {
+  PostSubmissionAssessment,
+  PriorApprovalAssessment,
+} from "@/types/odp-types/schemas/postSubmissionApplication/data/Assessment";
+import { getDescription } from "./application";
+
+/**
+ * Checks if the given object is a DprApplication.
+ * Run this before convertToDprApplication()
+ */
+export function isDprApplication(
+  app: DprPlanningApplication | DprApplication,
+): app is DprApplication {
+  // Return `true` only if all required DprApplication fields exist
+  return (
+    !!app &&
+    "data" in app &&
+    !!app.data &&
+    "application" in app.data &&
+    !!app.data.application &&
+    "reference" in app.data.application
+  );
+}
+
+/**
+ * Checks to see if we're in the consultation period
+ * @param startDate string
+ * @param endDate string
+ * @returns boolean
+ */
+export const getIsConsultationPeriod = (
+  startDate: Date,
+  endDate: Date,
+): boolean => {
+  const now = new Date();
+
+  return now >= startDate && now <= endDate;
+};
+
+export const convertToDprApplication = (
+  app: DprPlanningApplication,
+): DprApplication => {
+  let stage = undefined;
+  let status = undefined;
+  let withdrawnAt = undefined;
+  const withdrawnReason = undefined;
+  let consultation = undefined;
+
+  const isConsultationPeriod =
+    app.application.consultation.startDate &&
+    app.application.consultation.endDate
+      ? getIsConsultationPeriod(
+          new Date(app.application.consultation.startDate),
+          new Date(app.application.consultation.endDate),
+        )
+      : false;
+
+  if (app.application.status === "closed") {
+    status = "in_assessment";
+  }
```

Comment:
Maybe just add a comment here to say why we're doing this? https://github.com/tpximpact/digital-planning-register/pull/290#discussion_r2025418159